packages/sdk/src/across/rateModel.ts

```diff
@@ -0,0 +1,174 @@
+import { expectedRateModelKeys, RateModel } from "./constants";
+import { exists } from "../utils";
+import { ethers } from "ethers";
+
+// Each L1 token is mapped to an array of stringified rate models, ordered by the block height at which they were
+// published on-chain. This dictionary is used internally to fetch a rate model for a block height.
+type RateModelEventsDictionary = {
+  [l1TokenAddress: string]: { blockNumber: number; rateModel: string }[];
+};
+
+// Events should be reformatted into this shape to be used as input into methods in this file.
+export type RateModelEvent = {
+  blockNumber: number;
+  transactionIndex: number;
+  logIndex: number;
+  rateModel: string;
+  l1Token: string;
+};
+
+export class RateModelDictionary {
```

Comment:
Should we add tests for this class?

Reply:
[Added tests](https://github.com/UMAprotocol/protocol/pull/3673/commits/bae72c18a055027d7f048c848e43d65167794df1)